src/Task.ts

```diff
@@ -522,6 +542,11 @@ export class Task {
         const newTasks: Task[] = [];
 
         if (nextOccurrence !== null) {
+            const { setCreatedDate } = getSettings();
+            let createdDate: moment.Moment | null = null;
+            if (setCreatedDate) {
+                createdDate = window.moment();
+            }
```

Comment:
Thank you for fixing the created dates with toggling.

I tested it and it worked well.

This will be an important set of functionality to get under automated tests, and it will be a bit of a pain because of the number of combinations to test.

| what                                   | Setting to add created date is on                              | Setting to add created date is off  |
| -------------------------------------- | -------------------------------------------------------------- | ----------------------------------- |
| Initial Task had created date          | New task has created date for the date the task was toggled on | New task does not have created date |
| Initial Task did not have created date | New task has created date for the date the task was toggled on | New task does not have created date | 

I looked at the existing tests in Task.test.ts for recurrence, and I don't think this set really fits in to it.

So I think it will be best to create some new tests specifically, at the end of the `toggle done` group of tests in `Task.test.ts`

I don't know how many tests you have written in the past.

If you are new to testing, this is probably quite a big ask - feel free to leave it for me to do, if you wish.

Reply:
I wrote all four cases in tests.